### DISCUSSION TITLE
Substituion filter with RegEx - don't understand the results

### DISCUSSION THREAD

**Discussion by Pierre-de-la-Verre** [WinMerge/winmerge#1963] [General]
My idea for the following definition is: ignore all line beginning with "job" and all lines beginning with "input file". So I defined "job.+" as "job" as string plus "." for everything plus "+" for repeat until the end.
But it seems that both definition are depending on each other - when I activate only "job", the result is wrong. Activating bot "job" and "input file", the result is OK.
What's wrong here?
![winmerge_a](https://github.com/WinMerge/winmerge/assets/13277944/3fcc2302-7a49-4f50-9f66-7aed2e16bf73)

![winmerge_b](https://github.com/WinMerge/winmerge/assets/13277944/40070be8-5487-4d98-853a-cb3d3e18a020)

**Comment by sdottaka** [comment]
This is something that I think needs improvement, but I haven't been able to address it yet.
Substitution filters are applied per difference block instead of per difference line.
If the result of applying Substitution filters is that some lines in a difference block match but other lines do not match, the differences in that difference block are not ignored on a block-by-block basis.

A PrediffLineFilter plugin with similar functionality should give better results than the Substitutuion filters.

![image](https://github.com/WinMerge/winmerge/assets/98126/f5d4a057-4f81-456f-890b-7dac2b5481d3)
![image](https://github.com/WinMerge/winmerge/assets/98126/f19cc769-9e61-4209-9277-b03ccda37eb9)
![image](https://github.com/WinMerge/winmerge/assets/98126/e04e7454-8b5e-4a87-b7d0-7d99a4cee8a8)

**Comment by Pierre-de-la-Verre** [comment]
Thanks @sdottaka for your (as usual) detailed answer.
> ...Substitution filters are applied per difference block instead of per difference line. 

Maybe you can add this information to the referring dialogue? This is an important thing.
